hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -550,6 +631,17 @@ public AbfsRestOperation checkAccess(String path, String rwx)
     return op;
   }
 
+  private void appendSASTokenToQuery(String path, String operation, AbfsUriQueryBuilder queryBuilder) throws SASTokenProviderException {
+    try {
+      LOG.trace("Fetch SAS token for {}", path);
+      String sasToken = sasTokenProvider.getSASToken(this.accountName, this.filesystem, path, operation);
+      queryBuilder.setSASToken(sasToken);
+      LOG.trace("SAS token fetch complete for {}", path);
+    } catch (Exception ex) {
+      throw new SASTokenProviderException("Failed to acquire a SAS token.", ex);
```

Comment:
Always preserve/add as much to the error string as you can. Especially as its all too easy for nested exceptions to not get logged on deep enough chains (i.e. hive)

add path and token to exception text, as well as ex.toString().

Reply:
Added